wpilibc/src/main/native/cpp/Error.cpp

```diff
@@ -16,21 +16,37 @@
 
 using namespace frc;
 
-void Error::Clone(const Error& error) {
-  m_code = error.m_code;
-  m_message = error.m_message;
-  m_filename = error.m_filename;
-  m_function = error.m_function;
-  m_lineNumber = error.m_lineNumber;
-  m_originatingObject = error.m_originatingObject;
-  m_timestamp = error.m_timestamp;
+Error::Error(Code code, llvm::StringRef contextMessage,
+             llvm::StringRef fileName, llvm::StringRef function, int lineNumber,
+             const ErrorBase* originatingObject) {
+  Set(code, contextMessage, fileName, function, lineNumber, originatingObject);
+}
+
+bool Error::operator<(const Error& rhs) const {
+  if (m_code < rhs.m_code) {
+    return true;
+  } else if (m_message < rhs.m_message) {
+    return true;
+  } else if (m_fileName < rhs.m_fileName) {
+    return true;
+  } else if (m_function < rhs.m_function) {
+    return true;
+  } else if (m_lineNumber < rhs.m_lineNumber) {
+    return true;
+  } else if (m_originatingObject < rhs.m_originatingObject) {
+    return true;
+  } else if (m_timestamp < rhs.m_timestamp) {
+    return true;
+  } else {
+    return false;
+  }
 }
 
 Error::Code Error::GetCode() const { return m_code; }
 
 std::string Error::GetMessage() const { return m_message; }
 
-std::string Error::GetFilename() const { return m_filename; }
+std::string Error::GetFilename() const { return m_fileName; }
```

Comment:
I have to ask.. what's driving the `filename` to `fileName` rename?  `filename` is quite a common variable name, so `fileName` feels odd to me.